### PR TITLE
Fix: float32 bug in UTs

### PR DESCRIPTION
**Pull request by anyangml**

### CODE REVIEW

File: deepmd/pt/model/network/mlp.py
Comment (njzjz) [inline deepmd/pt/model/network/mlp.py:1]: This might be not correct, since the tensor is not converted back to the original precision after this module.

Comment (njzjz) [reply deepmd/pt/model/network/mlp.py:1]: You may check whether the output value is float64 or float32

Comment (anyangml) [reply deepmd/pt/model/network/mlp.py:1]: so the output should always be `float64` or whatever the precision  of `xx` originally is?

Comment (njzjz) [reply deepmd/pt/model/network/mlp.py:1]: The precision here should only be applied to the neural networks, but not the output (such as energy, which usually has lots of digit)

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**